### ISSUE TITLE
Feature request for pandoc markdown: embedding pdf and video files

### ISSUE THREAD

**Issue by juligreen** [jgm/pandoc#7181]
Hey! :)
I'm trying some things out to create a workflow with pandoc markdown that is similar to [org-noter](https://github.com/weirdNox/org-noter).
This means I want to reference locations inside of pdfs (and later video files as well) inside my documents.
This is already somewhat possible by using html and latex directly inside of markdown:
```
---
header-includes: |
    \usepackage{pdfpages}
---

# Test title

Have some test text

\```{=html}
<object data="loremipsum.pdf#page=2" type="application/pdf" width="700px" height="700px">
    <embed src="loremipsum.pdf#page=2">
        <p>PDF download: <a href="loremipsum.pdf">Download PDF</a>.</p>
    </embed>
</object>
\```

\```{=latex}
\includepdf[pages={2-3}]{loremipsum.pdf}
\```
```

I put a backslash in front of the triple backticks so they don't break the code block.

After converting this markdown document into pdf, it embeds pages 2-3 from the loremipsum.pdf into the newly generated pdf.
[output.pdf](https://github.com/jgm/pandoc/files/6216986/output.pdf)

The generated html has the complete pdf embeded, but starts at page 2.

output.html:
```
<h1 id="test-title">Test title</h1>
<p>Have some test text</p>
<object data="loremipsum.pdf#page=2" type="application/pdf" width="700px" height="700px">
    <embed src="loremipsum.pdf#page=2">
        <p>Referenced pages 2-3: <a href="loremipsum.pdf">Download PDF</a>.</p>
    </embed>
</object>
```
[loremipsum.pdf](https://github.com/jgm/pandoc/files/6216991/loremipsum.pdf)

This part: `<p>Referenced pages 2-3: <a href="loremipsum.pdf">Download PDF</a>.</p>` only gets displayed in case the browser does not support embedded pdfs.

Now I would like to have some syntax in pandoc markdown that can generate each of these blocks for their respective filetypes.
I would imagine the syntax to be something like this: 
`!embed[pdf-title](loremipsum.pdf:2-3)`
Or anything like it.

A similar thing might be possible for embedding video files in html:
```
<object data="project.mp4#t=66,71" type="video/mp4">
    <embed src="project.mp4#t=66,71">
        <p>Referenced time: 66-71<a href="project.mp4">Download mp4</a>.</p>
    </embed>
</object>
```
[And you even might be able to include these in pdf](https://tex.stackexchange.com/questions/89088/how-to-embed-video-and-animation-in-latex-and-latex-beamer-step-by-step), but that might be undesireable.

If this is a stupid idea, feel free to close this issue :wave:

**Comment by mb21** [comment]
For html output, this already works:

```
~ echo '![](foo.mov)' | pandoc
<p><video src="foo.mov" controls=""><a href="foo.mov">Video</a></video></p>

~ echo '![](foo.pdf)' | pandoc
<p><embed src="foo.pdf" /></p>
```

But not for LaTeX/PDF output AFAIK...

**Comment by juligreen** [comment]
Ohh that's awesome, but can you reference the pages/timestamps?

**Comment by juligreen** [comment]
After trying it out, it seems like it does!
```
echo '![](foo.pdf#page=2-3)' | pandoc 
<p><embed src="foo.pdf#page=2-3" /></p>
```

This still opens the pdf at the correct page.

**Comment by Polirecyliente** [comment]
@juligreen you said:

> [And you even might be able to include these in pdf](https://tex.stackexchange.com/questions/89088/how-to-embed-video-and-animation-in-latex-and-latex-beamer-step-by-step), but that might be undesireable.

This is not undesirable, quite the contrary, embedding videos in PDF would be useful just like embedding images in PDF.

**Comment by juligreen** [comment]
Though regarding video files, while it works correctly with ".mov" files, mkv get incorrectly recognized as images
```
~ echo '![](foo.mkv)' | pandoc
<p><img src="foo.mkv" /></p>
```

**Comment by mb21** [comment]
yeah, I guess we could add mkv [here](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/MIME.hs#L70)... and/or make a pull to happstack where we get that list from.

About adding support of videos to the default LaTeX template, I think if it requires uncommon packages, then we would not want to do that... as you can simply use a custom template if you need to...

**Comment by heygent** [comment]
Right now, I am able to embed the first page of a pdf with the image syntax, however I cannot specify the page. I think an easy way to implement pdf page embedding would be to pass the attribute `page` as is to the `\includegraphics` command, in a similar way done for the `width` and `height` attributes.

https://pandoc.org/MANUAL.html#extension-link_attributes

So that a specific pdf page could be embedded this way:

```
![Global frog population as of 1943.](slides.pdf){ page=13 }
```

Right now I am able to achieve what I want by putting in the raw latex command directly, but that comes at the cost of simple captioning. Plus, the `graphicx` package doesn't get loaded unless the `![]()` syntax is used, so it's necessary to put an image somewhere or to load the package manually in order for this to work.

```
\includegraphics[page=6]{../slide/Lecture03ns02.pdf}
```

**Comment by jgm** [comment]
So you can specify `page=6` and includegraphics will use that page of a pdf? I didn't know that. Yes, I think it would be straightforward to react this way to the `page` attribute if present.

**Comment by MinmoTech** [comment]
Just as a heads up, I think the mime-type of mkv is still not correctly set as mentioned here: https://github.com/jgm/pandoc/issues/7181#issuecomment-817095813

**Comment by jgm** [comment]
What should the mime type of .mkv be?

**Comment by mb21** [comment]
@jgm [Wikipedia says](https://en.wikipedia.org/wiki/Matroska) `video/x-matroska`.

**Comment by MinmoTech** [comment]
So the answer to that is not super easy.
This page shows container-browser compatibility: https://jellyfin.org/docs/general/clients/codec-support.html#container-compatibilityhttpsdevelopermozillaorgen-usdocswebmediaformatscontainers

Firefox and Safari are the only browsers without mkv support currently (though I'm not even certain about Safari)
I created an html file with the following contents:
```html
<p><video src="test.mkv" controls=""><a href="test.mkv">Video</a></video></p>
```
This plays without problems in chromium-based browsers, but shows the following for Firefox: 
![image](https://user-images.githubusercontent.com/25890405/160788362-a3accc78-ef80-4902-9f38-95e607544ced.png)

But it doesn't end there since mkv can hold any kind data inside which is especially problematic with some video and audio codecs, since they don't play at all in most browsers (the ones that work will show, the ones that don't will not, so you can have audio playing without video for example).

Btw adding `type="video/webm"` as the mime-type explicitly to the html element didn't change anything in my testing but it might it some rare edge-cases.


Even with all my ramblings here, I think the option should still be given to the user to embed mkv files if they know that they are compatible (and I pray for Firefox support in the meantime).


The thing I actually want is `echo '![](foo.mkv)' | pandoc` outputting `<p><video src="foo.mkv" controls=""><a href="foo.mkv">Video</a></video></p>` instead of `<p><img src="foo.mkv" /></p>` like it currently does

**Comment by mb21** [comment]
Thanks for the investigation! Adding `video/x-matroska` as the mime type in the pandoc source code will cause pandoc to output `<video src="foo.mkv"`... so we probably should just do that...
